### PR TITLE
fix: resolve failing test

### PR DESCRIPTION
**Pull request by james-a-morris**
This PR resolves all failing tests in the SDK refactor.

### CODE REVIEW

File: src/clients/SpokePoolClient.ts [hunk 1]
```diff
@@ -438,14 +438,7 @@ export class SpokePoolClient extends BaseAbstractClient {
...
   }> {
-    return getBlockRangeForDepositId(
-      targetDepositId,
-      initLow,
-      initHigh,
-      maxSearches,
-      this.spokePool as SpokePool,
-      this.deploymentBlock
-    );
+    return getBlockRangeForDepositId(targetDepositId, initLow, initHigh, maxSearches, this, this.deploymentBlock);
```
Comment (james-a-morris) [inline src/clients/SpokePoolClient.ts:441]: Passing the spoke client directly saves us an RPC call to the latest block number. It also allows us to leverage the mocked clients

File: test/SpokePoolClient.ValidateFill.ts
Comment (james-a-morris) [inline test/SpokePoolClient.ValidateFill.ts:1]: @nicholaspai @pxrl I've changed test logic. I've added a bunch of comments here, but please let me know if we need to huddle to drill down on any of this

File: src/utils/SpokeUtils.ts [hunk 2]
```diff
@@ -20,36 +21,14 @@ export async function getBlockRangeForDepositId(
...
-  // at the low block, and the deposit ID at the high block in parallel.
-  const [mostRecentBlockNumber, highestPossibleDepositIdInRange, lowestPossibleDepositIdInRange] = await Promise.all([
-    spokePool.provider.getBlockNumber(),
-    _getDepositIdAtBlock(initHigh),
-    _getDepositIdAtBlock(Math.max(deploymentBlock, initLow - 1)),
-  ]);
-
   // Set the initial high block to the most recent block number or the initial high block, whichever is smaller.
-  initHigh = Math.min(initHigh, mostRecentBlockNumber);
+  initHigh = Math.min(initHigh, spokePool.latestBlockNumber);
```
Comment (nicholaspai) [inline src/utils/SpokeUtils.ts:31]: `spokePool` is guaranteed to be `update`ed right?

**Review by nicholaspai — APPROVED** [review]
One 1

File: src/utils/SpokeUtils.ts (hunk 2 above)
Comment (james-a-morris) [reply src/utils/SpokeUtils.ts:31]: Good call - resolved

File: src/clients/SpokePoolClient.ts (hunk 1 above)
Comment (pxrl) [reply src/clients/SpokePoolClient.ts:441]: If we pass in the SpokePoolClient, then we could also drop the subsequent block number (`this.deploymentBlock`). wdyt?

File: src/utils/SpokeUtils.ts [hunk 3]
```diff
@@ -72,14 +53,36 @@ export async function getBlockRangeForDepositId(
...
+  const queriedIds: Record<number, number> = {};
+
+  // Define a llambda function to get the deposit ID at a block number. This function will first check the
+  // queriedIds cache to see if the deposit ID at the block number has already been queried. If not, it will
+  // make an eth_call request to get the deposit ID at the block number. It will then cache the deposit ID
+  // in the queriedIds cache.
+  const _getDepositIdAtBlock = async (blockNumber: number): Promise<number> => {
+    if (queriedIds[blockNumber] === undefined) {
+      queriedIds[blockNumber] = await spokePool._getDepositIdAtBlock(blockNumber);
+    }
```
Comment (pxrl) [inline src/utils/SpokeUtils.ts:66]: Possible concise alternative ?

```suggestion
    queriedIds[blockNumber] ??= await spokePool._getDepositIdAtBlock(blockNumber);
```

File: src/utils/SpokeUtils.ts [hunk 4]
```diff
@@ -72,14 +53,36 @@ export async function getBlockRangeForDepositId(
...
   // If the deposit ID at the initial high block is less than the target deposit ID, then we know that
   // the target deposit ID must be greater than the initial high block, so we can throw an error.
   if (highestPossibleDepositIdInRange <= targetDepositId) {
     // initLow   = 5: Deposits Num: 10
     //                                     // targetId = 11  <- fail (triggers this error)          // 10 <= 11
     //                                     // targetId = 10  <- fail (triggers this error)          // 10 <= 10
     //                                     // targetId = 09  <- pass (does not trigger this error)  // 10 <= 09
-    throw new Error(`Target depositId is greater than the initial high block (${targetDepositId} > ${initHigh})`);
+    throw new Error(
+      `Target depositId is greater than the initial high block (${targetDepositId} > ${highestPossibleDepositIdInRange})`
```
Comment (pxrl) [inline src/utils/SpokeUtils.ts:84]: `highestPossibleDepositIdInRange` pushes us out over 120 chars here, and prettier unfortunately doesn't propose a refactor to fix that. Would you consider `highestDepositIdInRange` instead?

The error `Target depositId out of range (${targetDepositId} > ${highestDepositIdInRange})` might also help to consolidate it a bit.

File: test/SpokePoolClient.ValidateFill.ts [hunk 5]
```diff
@@ -242,21 +246,25 @@ describe("SpokePoolClient: Fill Validation", async function () {
...
-      spokePool1DeploymentBlock,
-      spokePoolClient1.latestBlockNumber,
-      10
+
+    expect(searchRange1.high).to.be.greaterThanOrEqual(deposit1Block);
+    expect(searchRange1.low).to.be.lessThanOrEqual(deposit1Block);
+
+    // Searching for deposit ID 2 that doesn't exist yet should throw.
+    await assertPromiseError(
+      spokePoolClient1._getBlockRangeForDepositId(2, spokePool1DeploymentBlock, spokePoolClient1.latestBlockNumber, 10),
```
Comment (pxrl) [inline test/SpokePoolClient.ValidateFill.ts:255]: Might be nice to grab the depositId emitted when the deposit was made - i.e. up around line 213, where `deposit1Block` is extracted.

File: src/clients/SpokePoolClient.ts (hunk 1 above)
Comment (james-a-morris) [reply src/clients/SpokePoolClient.ts]: Done

File: test/SpokePoolClient.ValidateFill.ts (hunk 5 above)
Comment (james-a-morris) [reply test/SpokePoolClient.ValidateFill.ts:255]: Could you expand on this? My understanding is that this call is meant to throw

**Pull request merged**